modules/backend/assets/js/october.tabformexpandcontrols.js

```diff
@@ -160,4 +160,209 @@
     }
 
     $.oc.tabFormExpandControls = TabFormExpandControls
-}(window.jQuery);
\ No newline at end of file
+
+}(window.jQuery);
+
+
+/*
+ * Auto update WAI-ARIA when a user updates the tabs
+ */
+ $(document).ready(function(){
+
+     /* Master Tabs */
+
+     // if no selected => select first
+     if ($('.master-tabs a[aria-selected]').length == 0) {
+         $('.master-tabs a:first').attr('aria-selected', 'true');
+         $('.master-tabs a:first').attr('tabindex', '0');
+     }
+
+     /* click on a link */
+     $('.master-tabs a').click(function() {
+         // remove aria selected on all links + remove focusable
+         $('.master-tabs a').attr('aria-selected', 'false');
+
+         // add aria selected on $this + focusable
+         $('.master-tabs a.focus-visible').attr('aria-selected', 'true');
+
+         // Important - Must be set to true for October to set class="active" in the <li>
+         return true;
+     });
+
+     $('body').on('keydown','.master-tabs', function(event) {
+         // strike up or left in the tab
+         if (event.keyCode == 37 || event.keyCode == 38) {
```

Comment:
@ayumihamsaki you could simplify this code block to something like that (quick try):

```
let strikeUpOrRightTab = event.keyCode == 37 || event.keyCode == 38;
let strikeDownOrLeftTab = event.keyCode == 40 || event.keyCode == 39;
if (strikeUpOrRightTab || strikeDownOrLeftTab) {
    event.preventDefault();

    var postition = strikeUpOrRightTab ? 'first-child' : 'last-child';
    $activated = $('.master-tabs a[aria-selected="true"]').parent();
    if ($activated.is('.master-tabs li:' + position)) {
        $('.master-tabs li:' + position +' a').click().focus();
    } else {
        // else activate previous
        $activated.prev().children('.master-tabs a').click().focus();
    }
}
```

Reply:
@w20k Need to talk to you on this part, please have a read: https://stackoverflow.com/questions/905222/enter-key-press-event-in-javascript/905233#905233

This section:

> NOTE:
> It's been pointed out that keyCode is now deprecated. The next best alternative which has also been deprecated.
> 
> Unfortunately the favored standard key, which is widely supported by modern browsers, has some dodgy behavior in IE and Edge. Anything older than IE11 would still need a polyfill.
> 
> Furthermore, while the deprecated warning is quite ominous about keyCode and which, removing those would represent a massive breaking change to untold numbers of legacy websites. For that reason, it is unlikely they are going anywhere anytime soon.

Reply:
Shipped in all browsers except Safari, but needs a polyfill to cover Edge and IE, https://www.chromestatus.com/feature/4748790720364544

Reply:
@ayumihamsaki if you're talking about keyCode, I'm aware that they are deprecated. Make sense, though as it stated they won't be removed in few decades for sure 😄 

You'd still need `preventDefault` or `return false`, depending on what you really want only prevent event or both.
I was talking more about this thing: https://stackoverflow.com/a/1357151/1053650 also this one https://web.archive.org/web/20160429070956/http://fuelyourcoding.com/jquery-events-stop-misusing-return-false/

Reply:
I want change them to event.key just want to know where to stick the polyfill ? It would be useful for the rest of the backend as well, to use the polyfill.

Did some searching and found the bug here: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/11509049/

Reply:
It's worth saying that most people using 'Screen Readers' use Chrome, they say the other browser screen readers are not nice and they tend to stick to the same screen reader.

Reply:
Clicking resolved on this, see my latest commits - have compressed the code down and moved to `event.key` now.